R/functions.R

```diff
@@ -178,14 +196,27 @@ get_biomart <- function(gene_ids, host, organism) {
     return(biomart_results)
   }
 }
+#'Duplicate HGNC
+#'
+#'Count normalization requires Ensembl Ids to be unique. In rare cases, there are more
+#'than one HGNC symbol per gene Id. This function collapses the duplicate entries into
+#'a single entry by appending the HGNC symbols in a comma separated list.
+#'@param biomart_results Output of `get_biomart()`
+#'
+collapse_duplicate_hgnc_symbol <- function(biomart_results){
+  biomart_results %>%
+    dplyr::group_by(ensembl_gene_id) %>%
+    dplyr::mutate(hgnc_symbol = paste(hgnc_symbol, collapse = ", ")) %>%
+    unique()
+}
 #'Filter genes
 #'
 #'Remove genes that have less than 1 counts per million (cpm) in at least 50%
 #'of samples per condition. If a biomaRt object is provided, gene lengths and
 #'gene GC content is required and genes with missing values are also removed.
 #'
 #'@inheritParams md
-#'@param count_df A counts data frame with sample identifiers as rownames.
+#'@inheritParams count_df
 #'
 filter_genes <- function(md, count_df) {
   genes_to_analyze <- md %>%
```

Comment:
##Add code to handle issues with wills pipeline
count_df  <- count_df[ (row.names(count_df) %in% c("N_unmapped", "N_multimapping", "N_noFeature", "N_ambiguous")) == F, ]
##Also Throw Warning to account for other aligners
## This only works for ENSGs E could probably come up with a scaleable method for Gene Symbols
if (!all(grepl("ENSG", row.names(count_df)))) {
      mismatched <- row.names(count_df)[ grepl("ENSG", row.names(count_df) == F ]
      stop(glue::glue("Features {mismatched} are not ENSGs."))
}